lib/arborist/build-ideal-tree.js

```diff
@@ -355,6 +359,41 @@ module.exports = cls => class IdealTreeBuilder extends cls {
     })
   }
 
+  async [_retrieveSpecName] (spec) {
+    // if it's just @'' then we reload whatever's there, or get latest
+    // if it's an explicit tag, we need to install that specific tag version
+    const isTag = spec.rawSpec && spec.type === 'tag'
+
+    if (spec.name && !isTag)
+      return spec
+
+    const mani = await pacote.manifest(spec)
+    // if it's a tag type, then we need to run it down to an actual version
+    if (isTag)
+      return npa(`${mani.name}@${mani.version}`)
+
+    spec.name = mani.name
+    return spec
+  }
+
+  async [_followSymlinkPath] (spec, currpath) {
+    if(spec.type === 'directory') {
+      const real = await (
+        realpath(spec.fetchSpec, this[_rpcache], this[_stcache])
+          // TODO: create synthetic test case to simulate realpath failure
+          .catch(/* istanbul ignore next */() => null)
```

Comment:
I couldn't get creative enough to come up with a test that would allow us to test for the `realpath` throw without creating a huge mess, I think is fine to leave it as a TODO for later 😬